src/pip/_internal/cli/parser.py

```diff
@@ -21,54 +29,59 @@
+        r"`(?P<syntax>[^`]*)`",  # highlight `text in backquotes` as syntax
+    ]
+
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         # help position must be aligned with __init__.parseopts.description
         kwargs["max_help_position"] = 30
         kwargs["indent_increment"] = 1
         kwargs["width"] = shutil.get_terminal_size()[0] - 2
         super().__init__(*args, **kwargs)
-
-    def format_option_strings(self, option: optparse.Option) -> str:
-        return self._format_option_strings(option)
-
-    def _format_option_strings(
-        self, option: optparse.Option, mvarfmt: str = " <{}>", optsep: str = ", "
-    ) -> str:
-        """
-        Return a comma-separated list of option strings and metavars.
-
-        :param option:  tuple of (short opt, long opt), e.g: ('-f', '--format')
-        :param mvarfmt: metavar format string
-        :param optsep:  separator
-        """
-        opts = []
-
-        if option._short_opts:
-            opts.append(option._short_opts[0])
-        if option._long_opts:
-            opts.append(option._long_opts[0])
-        if len(opts) > 1:
-            opts.insert(1, optsep)
-
-        if option.takes_value():
-            assert option.dest is not None
-            metavar = option.metavar or option.dest.lower()
-            opts.append(mvarfmt.format(metavar.lower()))
-
-        return "".join(opts)
+        # This is unfortunate but necessary since arguments may have not been
+        # parsed yet at this point, so detect --no-color manually.
+        no_color = (
+            "--no-color" in sys.argv
+            or bool(strtobool(os.environ.get("PIP_NO_COLOR", "no") or "no"))
+            or "NO_COLOR" in os.environ
+        )
+        self.console = PipConsole(theme=Theme(self.styles), no_color=no_color)
+        self.rich_option_strings: dict[optparse.Option, Text] = {}
+
+    def stringify(self, text: RenderableType) -> str:
+        """Render a rich object as a string."""
+        with self.console.capture() as capture:
+            self.console.print(text, highlight=False, soft_wrap=True, end="")
+        help = capture.get()
+        return "\n".join(line.rstrip() for line in help.split("\n"))
 
     def format_heading(self, heading: str) -> str:
         if heading == "Options":
             return ""
-        return heading + ":\n"
+        rich_heading = Text().append(heading, "optparse.groups").append(":\n")
+        return self.stringify(rich_heading)
 
     def format_usage(self, usage: str) -> str:
         """
         Ensure there is only one newline between usage and the first heading
         if there is no description.
         """
-        msg = "\nUsage: {}\n".format(self.indent_lines(textwrap.dedent(usage), "  "))
-        return msg
+        rich_usage = (
+            Text("\n")
+            .append("Usage", "optparse.groups")
+            .append(f": {self.indent_lines(textwrap.dedent(usage), '  ')}\n")
```

Comment:
I suggest using the same colour for the colon:


<img width="190" height="113" alt="image" src="https://github.com/user-attachments/assets/16a78afd-31a8-43f0-acf4-2f7305dbde47" /> -> <img width="199" height="112" alt="image" src="https://github.com/user-attachments/assets/613020fd-d084-4208-9615-1dd623584a54" />


```suggestion
            .append("Usage:", "optparse.groups")
            .append(f" {self.indent_lines(textwrap.dedent(usage), '  ')}\n")
```
(and elsewhere)